app/models/container_node.rb

```diff
@@ -77,6 +77,14 @@ def ipaddress
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    address = ipaddress || name
+    if MiqRegion.my_region.role_active?("cockpit_ws")
+      uri = URI(MiqCockpit::WS::MIQ_REDIRECT_URL)
+      uri.query = {:ip => address }.to_query
```

Comment:
Similar to the other comments, this is pushing the logic into apache's configuration instead of using the well known active cockpit appliance.